### ISSUE TITLE
Base and reference model always generating the same logits

### ISSUE THREAD

**Issue by israaexol** [CarperAI/trlx#359]
### 🐛 Describe the bug

Hello,

I'm facing an issue with the base `(self.model)` and reference model `(self.model.forward_hydra)`, they're constantly generating the same logits for all samples and therefore, the KL divergence (as shown in the picture below) is always 0, making it impossible to move to the next step.
![kl div](https://user-images.githubusercontent.com/47036943/223468038-7e789867-d3d7-41db-b183-abba41648dc3.PNG)
Instead of the default OpenAI GPT, I'm using a fine-tuned model of AraGPT2 (GPT2 that supports Arabic), with its respective reward model of course.
Any idea on why this could be happening? Aren't they supposed to be different according to the implementation of the forward_hydra function?
I tried to change multiple parameters related to the reference model but with no use
Thanks in advance!

### Additional system and package information

Paperspace VM
